modules/storage-api/src/testFixtures/java/org/apache/ignite/internal/storage/index/impl/TestSortedIndexStorage.java

```diff
@@ -45,7 +44,9 @@
  * Test implementation of MV sorted index storage.
  */
 public class TestSortedIndexStorage implements SortedIndexStorage {
-    private final ConcurrentNavigableMap<ByteBuffer, Set<RowId>> index;
+    private static final Object NULL = new Object();
+
+    private final ConcurrentNavigableMap<ByteBuffer, NavigableMap<RowId, Object>> index;
```

Comment:
It looks like this can still be `ConcurrentNavigableMap<ByteBuffer, NavigableSet<RowId>>` which is simpler to use and it causes less confusion (because something that is used as a set is called Set, not Map)

Reply:
Main problem in `java.util.SortedSet#first` if the set is empty it will throw an exception.

Reply:
Well, I changed the code to migrate it from maps to sets, and tests passed. This is probably because the set/map is never empty (if we remove the last element of it, we remove the set/map from the `index`). So the fact that `SortedSet#first()` throws for an empty set does not seem to be a problem.

Reply:
If we look at `TestSortedIndexStorage#remove`, we can get into trouble if, after getting "RowId"s, in parallel delete first element (become empty RowIds) it and then try to take the first element.

Reply:
Yes, you are right. But we can just wrap the call to `set.first()` with `try`/`catch NoSuchElementException`. This needs to be done in just 2 places, and we'll have a set represented by a Set, not by a Map. Still less confusion.

Reply:
It's not good to write code based on exceptions, it's better to leave the map.

Reply:
It's ok to use any way to pass information. In this case, the library chose to use exceptions to this end, so we are ok to catch them. This is a little ugly, but it's a lot less ugly than using a map as a set.

If you are going to leave the map, please at least add a comment to the field to explain that the map is actually used to model a set and why it is done so.

Reply:
You are right, I should write about it in the comments.